### ISSUE TITLE
A question of quantile loss

### ISSUE THREAD

**Issue by Wang-XR** [dmlc/xgboost#9756]
I haven't used C language before, so I can't understand the code in "src/objective/quantile_obj.cu".
I want to know if multiple XGBoost models will be established to fit different quantiles when I input multiple quantiles? Or perhaps there is only one XGBoost model, whose loss function is the simple average of the loss functions for all quantiles.

I found a phenomenon where when I input multiple quantiles [0.01,0.02,..., 0.98,0.99] into the model, the 1% quantile model is labeled as Model 1; When I input quantile 0.01 into the model, the 1% quantile model is labeled as Model 2, and the results of Model 2 and Model 1 differ significantly. So I infer that when multiple quantiles are input into the model, the loss function is the simple average of the loss functions for all quantiles. Is this inference correct?

I find the function "check_quantile_loss" in "python-package/xgboost/testing/updater.py" is also used to verify the problem I raised, but it has not been used in other files.

This question will help with some comparative experiments, and we look forward to your answer. Thank you very much!

**Comment by trivialfis** [comment]
At the moment, we are using one model for each quantile. Under the hood, it's the same as multi output regression. That said, we will support using vector leaf in the future.

**Comment by Wang-XR** [comment]
> At the moment, we are using one model for each quantile. Under the hood, it's the same as multi output regression. That said, we will support using vector leaf in the future.
Thank you for your answer, but I still have some questions.
Example 1
`
Xy = xgb.QuantileDMatrix(x_train, y_train)
Xy_test = xgb.QuantileDMatrix(x_test, y_test)
evals_result={}
alpha = np.linspace(0.01, 0.99, 99)
booster = xgb.train(
        {
            "objective": "reg:quantileerror",
            "quantile_alpha": alpha,
            "learning_rate": 0.01,
            "max_depth": 6,
        },
        Xy,
        num_boost_round=1000,
        evals=[(Xy, "Train"), (Xy_test, "Test")],
        evals_result=evals_result,
    )
scores1 = booster.predict(xgb.DMatrix(x_train))
scores1[1]
`
Example 2
`Xy = xgb.QuantileDMatrix(x_train, y_train)
Xy_test = xgb.QuantileDMatrix(x_test, y_test)
evals_result={}
booster = xgb.train(
        {
            "objective": "reg:quantileerror",
            "quantile_alpha": 0.01,
            "learning_rate": 0.01,
            "max_depth": 6,
        },
        Xy,
        num_boost_round=1000,
        evals=[(Xy, "Train"), (Xy_test, "Test")],
        evals_result=evals_result,
    )
scores1 = booster.predict(xgb.DMatrix(x_train))
scores1`
There is a significant difference in the results of the 1% quantile regression between Example 1 and Example 2.
When we input vectors [0.01,0.02,..., 0.99], if each quantile corresponds to a tree model and its loss function, then their results should be consistent with those obtained by inputting scalars separately (such as 0.01).
I don't know what caused this difference.

**Comment by trivialfis** [comment]
We fit a single intercept for all models at the moment. You can try to hardcode the `base_score` parameter to some value (like 0).

Aside from that, please don't create the test matrix like this:
``` python
Xy_test = xgb.QuantileDMatrix(x_test, y_test)
```

You need the `ref` parameter as well
```
Xy_test = xgb.QuantileDMatrix(x_test, y_test, ref=Xy_train)
```

**Comment by Wang-XR** [comment]
Thank you for your answer！It should be the difference caused by the random initialization of 'base_score'.

**Comment by trivialfis** [comment]
Feel free to reopen.
